src/Services/InsuranceCompanyService.php

```diff
@@ -214,7 +219,9 @@ public function getInsuranceClaimTypes()
 
     public function insert($data)
     {
-        $freshId = $this->getFreshId("id", "insurance_companies");
+        // insurance companies need to use sequences table since they share the
+        // addresses table with pharmacies
+        $freshId = generate_id();
```

Comment:
Of the two ways to go, this is the best imo and obviously yours.

Reply:
thanks for reviewing and the advice this weekend!